utilities/virt.py

```diff
@@ -1006,26 +1012,46 @@ def _sc_name_for_storage_api():
         return (
             self.data_volume.pvc.instance.spec.accessModes
             if self.data_volume
-            else self.pvc.instance.spec.accessModes
-            if self.pvc
-            else self.data_volume_template["spec"][api_name].get("accessModes")
-            or StorageProfile(name=_sc_name_for_storage_api()).instance.status["claimPropertySets"][0]["accessModes"]
+            else (
+                self.pvc.instance.spec.accessModes
+                if self.pvc
+                else self.data_volume_template["spec"][api_name].get("accessModes")
+                or StorageProfile(name=_sc_name_for_storage_api()).instance.status["claimPropertySets"][0][
+                    "accessModes"
+                ]
+            )
         )
 
     @property
     def virtctl_port_forward_cmd(self):
         return f"{VIRTCTL} port-forward --stdio=true vm/{self.name}.{self.namespace} {SSH_PORT_22}"
 
-    @property
-    def login_params(self):
-        return py_config["os_login_param"][self.os_flavor]
+    def set_login_params(self):
+        _login_params = deepcopy(py_config["os_login_param"][self.os_flavor])
+
+        if not (self.username and self.password):
+            self.username = _login_params["username"]
+            self.password = _login_params["password"]
+
+            # Do not modify the defaults to OS like Windows where the password is already defined in the image
+            if self.os_flavor not in FLAVORS_EXCLUDED_FROM_CLOUD_INIT:
+                if self.exists:
+                    if cloud_init := [
```

Comment:
nit: a VM can start without cloudinitnocloud. should we consider that case too?

Reply:
do you mean when it is in the image?
can you share an example please (not windows)?

Reply:
I don't think we need to consider it, since the image will have it's own password/user and we have no way of knowing they're values, unless you know of a way.

Reply:
@rnetser here is a VM snippet : https://gitlab.cee.redhat.com/-/snippets/10067/raw/main/snippetfile1.txt